File: 프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.java
import java.util.*;

class Solution {
    public int solution(int[] queue1, int[] queue2) {
        int answer = -2;
        
        Queue<Long> q1 = new LinkedList<>();
        Queue<Long> q2 = new LinkedList<>();
        for(int i = 0; i < queue1.length; i++){
            q1.offer((long)queue1[i]);
            q2.offer((long)queue2[i]);
        }
        
        long sum1 = Arrays.stream(queue1).asLongStream().sum();
        long sum2 = Arrays.stream(queue2).asLongStream().sum();
        
        int cnt = 0;
        while(cnt < queue1.length * 3){
            if(sum1 > sum2){
                long v = q1.poll();
                sum1 -= v;
                sum2 += v;
                q2.offer(v);
            } else if(sum2 > sum1){
                long v = q2.poll();
                sum1 += v;
                sum2 -= v;
                q1.offer(v);
            } else return cnt;
            cnt++;
        }
        return (sum1 == sum2)? cnt : -1;
    }
}

File: 프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.py
from collections import deque

def solution(queue1, queue2):
    answer = -2
    
    cnt = 0
    dq1 = deque()
    dq2 = deque()
    sum1 = 0
    sum2 = 0
    for i in range(len(queue1)) :
        dq1.append(queue1[i])
        dq2.append(queue2[i])
        sum1 += queue1[i]
        sum2 += queue2[i]
        
    while(sum1 != sum2) :
        if cnt > len(queue1)*3 :
            return -1
        
        if(sum1 > sum2) :
            v = dq1.popleft()
            dq2.append(v)
            sum1 -= v
            sum2 += v
            
        elif(sum1 < sum2) :
            v = dq2.popleft()
            dq1.append(v)
            sum1 += v
            sum2 -= v
        cnt+=1        
    return cnt